props/websubmit/src/main.rs

```diff
@@ -177,6 +177,112 @@ fn run_sc_policy(ctx: Arc<Context>) -> Result<()> {
     })
 }
 
+/// If sensitive data is released, the release must be scoped, and all inputs to the scope must be safe.
+pub struct AuthDisclosureProp {
+    cx: Arc<PolicyContext>,
+}
+
+impl AuthDisclosureProp {
+    pub fn new(cx: Arc<PolicyContext>) -> Self {
+        AuthDisclosureProp { cx }
+    }
+
+    pub fn check(self) {
+        for c_id in self.cx.desc().controllers.keys() {
+            // All srcs that have no influencers
```

Comment:
I think this could be a `Context::roots(&self, edge_type: EdgeType)` primitive.

Reply:
Hm, would this primitive be useful in other policies, enough to warrant it's own primitive? Doing so would be easy but I was wondering why?

Reply:
done